### ISSUE TITLE
`io.load_results_from_dir` calls `ProtocolResults.from_dir` with unexpected keyword argument

### ISSUE THREAD

**Issue by robpkelly** [pyGSTio/pyGSTi#132]
**Describe the bug**
Loading a `ProtocolResults` object using `pygsti.io.load_results_from_dir` fails, raising `TypeError: from_dir() got an unexpected keyword argument 'quick_load'`

**To Reproduce**
Run the following demonstration script:
```python
import pygsti
from pygsti.modelpacks import smq1Q_XYI as smq
ds = pygsti.construction.generate_fake_data(smq.target_model(), smq.get_gst_circuits(1), 1000)
results = pygsti.do_long_sequence_gst(ds, smq.target_model(),
                                      smq.prep_fiducials(), smq.meas_fiducials(),
                                      smq.germs(), [1], verbosity=0)
results.write(dirname='/tmp/results')
loaded_results = pygsti.io.load_results_from_dir('/tmp/results')
```

The call to `pygsti.io.load_results_from_dir` on the last line raises with the following stack trace:

```
Traceback (most recent call last):
  File "local/bug_report.py", line 8, in <module>
    loaded_results = pygsti.io.load_results_from_dir('/tmp/results')
  File "/[...]/pyGSTi/pygsti/io/loaders.py", line 423, in load_results_from_dir
    return cls.from_dir(dirname, quick_load=quick_load)
  File "/[...]/pyGSTi/pygsti/protocols/protocol.py", line 1547, in from_dir
    dirname, pth.name, preloaded_data=data, quick_load=quick_load)
TypeError: from_dir() got an unexpected keyword argument 'quick_load'
```

**Expected behavior**
`pygsti.io.load_results_from_dir` should load and return a valid results instance without error.

**Environment (please complete the following information):**
 - pyGSTi version *0.9.9.2.post153+gf43dcda8*
 - python version *3.7.5*
 - OS *Linux 5.6.11-arch1-1*

**Additional context**
[The source line of the error](https://github.com/pyGSTio/pyGSTi/blob/2f3cbc2bb7e626b4aeb453cb6d3302b82aa92d60/pygsti/protocols/protocol.py#L1546). In the above example, the instance type returned by `io.cls_from_meta_json` is `protocols.gst.ModelEstimateResults` so we're ultimately calling [this classmethod](https://github.com/pyGSTio/pyGSTi/blob/2f3cbc2bb7e626b4aeb453cb6d3302b82aa92d60/pygsti/protocols/gst.py#L1250) which doesn't take a `quick_load` kwarg.

**Comment by robpkelly** [comment]
The quick-n-easy fix is to make sure any `from_dir` method or any method called in this way consumes unused keyword args. That's messy but so's life :shrug:

**Comment by enielse** [comment]
Fixed by a5d66b4a43ef8adb10c87c78a679cdec0ce38ba9.

(The `quick_load` argument should have been added to `ModelEstimateResults.from_dir` but must have been missed/forgotten when that enhancement was made a while ago.)

Let's leave this issue open until the above commit makes it into a release.

**Comment by enielse** [comment]
Version 0.9.10 fixes this issue.
